pkg/scalers/metrics_api_scaler.go

```diff
@@ -35,7 +68,29 @@ func NewMetricsAPIScaler(resolvedEnv, metadata, authParams map[string]string) (S
 	if err != nil {
 		return nil, fmt.Errorf("error parsing metric API metadata: %s", err)
 	}
-	return &metricsAPIScaler{metadata: meta}, nil
+
+	if meta.enableTLS {
+		config, err := kedautil.NewTLSConfig(meta.cert, meta.key, meta.ca)
+		if err != nil {
+			return nil, err
+		}
+
+		transport := &http.Transport{TLSClientConfig: config}
+		return &metricsAPIScaler{
+			metadata: meta,
+			client: &http.Client{
+				Timeout:   defaultTimeOut,
+				Transport: transport,
+			},
+		}, nil
+	}
+
+	return &metricsAPIScaler{
+		metadata: meta,
+		client: &http.Client{
+			Timeout: defaultTimeOut,
+		},
+	}, nil
```

Comment:
```suggestion
        client := &http.Client{
            Timeout:   defaultTimeOut,
        }
        
	if meta.enableTLS {
		config, err := kedautil.NewTLSConfig(meta.cert, meta.key, meta.ca)
		if err != nil {
			return nil, err
		}
		client.Transport = &http.Transport{TLSClientConfig: config}
	}

	return client, nil
```
I think may simplifies things a bit